app/registries/foreman/plugin/upgrade_task.rb

```diff
@@ -0,0 +1,35 @@
+module Foreman
+  class Plugin
+    class UpgradeTask
```

Comment:
Would Foreman ever want to use this? Would it be better then not under the plugin namespace?

Reply:
They may, and thats a good question.  I know the line has been "no new registries", and i see no where else to put this.  I could probably make it easier for foreman to define a task without defining a plugin.  Otherwise looking for guidance here from a foreman dev :)

Reply:
@ekohl @tbrisker @dLobatog some thoughts here?

Reply:
This may be useful for tasks in foreman as well, we often just push things into migrations that might not be ideal to run as such since we lack a different way to do so.

Reply:
@tbrisker I was kinda looking for a suggestion of how we would accomplish this, I could:

1.  Provide some easy way for foreman to add to this registry (seems kinda weird)
2.  Update the model to store the name and rake task and whether it has been run or not and rely on seeds to populate these values
3.  Create a new registry

I would normally say 2), but that makes defining order a little weird.  The nice part of using the plugin api, is that you the order is set based upon how they are defined by the plugin.